src/lsp/lp_doc.ml

```diff
@@ -12,6 +12,106 @@
 
 open Core
 module LSP = Lsp_base
+(*
+exception NoPosition of string *)
+
+module type RangeMap = sig
+
+  type point
+  type interval
+  type 'a t
+
+  val create_point : int -> int -> point
+  val l : point -> int
+  val c : point -> int
+
+  val interval_start : interval -> point
+  val interval_end : interval -> point
+
+  val create_interval : point -> point -> interval
+
+  type cmp = Before | In | After
+
+  val in_range : point -> interval -> cmp
+
+  val add : interval -> 'a -> 'a t -> 'a t
+  val find : point -> 'a t -> 'a option * interval
```

Comment:
Why isn't the output type ('a * interval) option?

Reply:
Fixed in the last pass.